addons/purchase_stock/models/stock.py

```diff
@@ -216,6 +217,27 @@ def action_view_purchase(self):
 
         return result
 
+    def action_replenish(self):
+        if self.route_id == self.env.ref('purchase_stock.route_warehouse0_buy'):
+            supplier = self.product_id.with_company(self.company_id.id)._select_seller(
+                                        quantity=self.product_min_qty,
+                                        uom_id=self.product_uom)
+
+            if not supplier and self.env['product.product'].check_access_rights('write'):
```

Comment:
Don't check the access write. Just bring it to the form view as readonly

Reply:
But what to check to open product form view in edit mode?